frontend/catalyst/jit.py

```diff
@@ -629,6 +639,17 @@ def capture(self, args):
                 self.user_function, static_argnums, abstracted_axes, full_sig, {}
             )
 
+            # After tracing ANY qjit jaxpr, add a transform_named_sequence primitive
+            # Otherwise -transform-interpreter will complain that a __transform_main is not found
```

Comment:
Inserting at the start of tracing indeed makes more sense.

Reply:
So before tracing, since the jaxpr is not produced yet, we cannot just append a new primitive to the list `jaxpr.eqns`, making inserting at the start of tracing a bit tricky. 

Thanks to @rmoyard 's idea, we make a new function right before tracing. The new function calls the `transform_named_sequence_p.bind()`, and then the user function. Now tracing this new function instead of the user function would produce exactly the same jaxpr except with a `transform_named_sequence` at the start!

Reply:
Nice!